### PR TITLE
updated tool "projective transformation" to support multi-channel and…

### PR DESCRIPTION
**Pull request by qgao-hd**
… RGB images

### CODE REVIEW

File: tools/projective_transformation/projective_transformation.py [hunk 1]
```diff
@@ -36,26 +47,65 @@ def warp_coords_batch(coord_map, shape, dtype=np.float64, batch_size=1000000):
...
-    warp_matrix = np.array(warp_matrix)
+def transform(moving_fn, fixed_fn, warp_mat, out):
 
-    trans = ProjectiveTransform(matrix=warp_matrix)
-    warped_coords = warp_coords_batch(trans, fixed_image.shape)
-    t = map_coordinates(moving_image, warped_coords, mode='reflect')
+    moving = skimage.io.imread(moving_fn)
+    extension = os.path.splitext(moving_fn)[1]
+    nDims = len(moving.shape)
+    assert nDims in [2, 3, 4, 5, 6], 'only support to 6D dimensions'
```
Comment (bgruening) [inline tools/projective_transformation/projective_transformation.py:55]: "We only support up to 6 dimensions"

**Review by bgruening — APPROVED** [review]
Otherwise, looks good to me!

File: tools/projective_transformation/projective_transformation.py (hunk 1 above)
Comment (qgao-hd) [reply tools/projective_transformation/projective_transformation.py]: Thanks for your prompt response. 

But the testing seemed failed. I had no idea.

**Comment by bgruening** [comment]
Test with `planemo test --biocontainers`

**Comment by qgao-hd** [comment]
> Test with `planemo test --biocontainers`

I tried planemo test on my local machine (Mac) with and without "--biocontainers" and both did not work. I got the error message "ModuleNotFoundError: No module named 'pandas'" from the test output, while there was no problem at all to directly run the python code. I even created a new (miniconda) environment and reinstalled Planemo exactly following the guide, but still got the same error.

On GitHub the error from the test output was different "ValueError: Could not find a backend to open `./out.dat`` with iomode `wi`". Here I could specify the format of output file because it is dependent on that of the input. I do not know how to deal with this issue. Making some random changes seems not a good idea...

**Comment by qgao-hd** [comment]
@bgruening Could you please have a look at the problem that I mentioned above. Thanks a lot!

**Comment by bgruening** [comment]
The error is:

```
Traceback (most recent call last):
  File "/home/runner/.planemo/planemo_tmp_x62rbmqz/projective_transformation.py", line 111, in <module>
    transform(args.moving_image, args.fixed_image, args.warp_matrix, args.out)
  File "/home/runner/.planemo/planemo_tmp_x62rbmqz/projective_transformation.py", line 101, in transform
    skimage.io.imsave(out + extension, warped_moving)
  File "/usr/local/lib/python3.9/site-packages/skimage/io/_io.py", line 136, in imsave
    return call_plugin('imsave', fname, arr, plugin=plugin, **plugin_args)
  File "/usr/local/lib/python3.9/site-packages/skimage/io/manage_plugins.py", line 207, in call_plugin
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/imageio/core/functions.py", line 195, in imwrite
    with imopen(uri, "wi", plugin=format) as file:
  File "/usr/local/lib/python3.9/site-packages/imageio/core/imopen.py", line 277, in imopen
    raise err_type(err_msg)
ValueError: Could not find a backend to open `./out.dat`` with iomode `wi`.
```

**Comment by bgruening** [comment]
My guess is that you need to pass the fileformat to the fuction or name ./out to ./out.png or ./out.jpeg etc ...

**Comment by qgao-hd** [comment]
> My guess is that you need to pass the fileformat to the fuction or name ./out to ./out.png or ./out.jpeg etc ...

This is also my guess. However the format of output is not fixed. It could be .tif, .png or .jpg, depending on the format of the input data. I read the Planemo docs and found flexible output format is allowed with the auto_format="true" attribute. But I did not know how to fix the problem.

**Comment by bgruening** [comment]
And an if else condition and provide different output filenames.

**Comment by qgao-hd** [comment]
> And an if else condition and provide different output filenames.

Thanks for the hint. I will have a try later today.

**Comment by qgao-hd** [comment]
All checks have passed. Commits will be merged.

**Pull request merged**